crates/oxc_prettier/src/doc.rs

```diff
@@ -31,15 +31,10 @@ pub enum Doc<'a> {
     /// no matter if the expression fits on one line or not.
     Hardline,
     /// Print something if the current `group` or the current element of `fill` breaks and something else if it doesn't.
-    IfBreak(Vec<'a, Doc<'a>>),
+    IfBreak(Box<Doc<'a>>),
```

Comment:
This should use `Box` from the allocator.

The whole IR should be allocated on the allocator. I'll fix it after your merge.